### PR TITLE
Added check on query result

### PR DESCRIPTION
**Pull request by dtrabo**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [x] Did you check UI tests on the sample app? They are not executed on CI.
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Added check on the size of the array returned by `executeSelectionQuery`

## Related PRs or issues

#1296

### CODE REVIEW

File: AppCenter/AppCenter/Internals/Storage/MSDBStorage.m [hunk 1]
```diff
@@ -126,7 +126,10 @@ + (void)setVersion:(NSUInteger)version inOpenedDatabase:(void *)db {
 
 + (void)enableAutoVacuumInOpenedDatabase:(void *)db {
   NSArray<NSArray *> *result = [MSDBStorage executeSelectionQuery:@"PRAGMA auto_vacuum" inOpenedDatabase:db];
-  int vacuumMode = [(NSNumber *)result[0][0] intValue];
+  int vacuumMode = 0;
+  if ([result count] != 0){
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Storage/MSDBStorage.m]:
```suggestion
  if ([result count] > 0 && [result[0] count] > 0) {
```

File: AppCenter/AppCenterTests/MSDBStorageTests.m [hunk 2]
```diff
@@ -169,6 +169,115 @@ - (void)testMigration {
...
+  OCMStub([dbStorageMock executeSelectionQuery:[OCMArg any] inOpenedDatabase:db]).andReturn(entries);
+  
+  // When
+  [MSDBStorage enableAutoVacuumInOpenedDatabase:db];
+  
+  // Then
+  OCMVerify([dbStorageMock executeNonSelectionQuery:[OCMArg any] inOpenedDatabase:db]);
+}
+
+
```
Comment (MatkovIvan) [inline AppCenter/AppCenterTests/MSDBStorageTests.m]: Extra line

File: AppCenter/AppCenterTests/MSDBStorageTests.m [hunk 3]
```diff
@@ -169,6 +169,115 @@ - (void)testMigration {
...
+  
+  // When
+  counter = [MSDBStorage getPageCountInOpenedDatabase:db];
+  
+  // Then
+  assertThatLong(counter, equalToLong(0));
+}
+
+- (void)testGetPageSizeInOpenedDatabaseReturnsZeroWhenQueryFails {
+  // If
```
Comment (MatkovIvan) [inline AppCenter/AppCenterTests/MSDBStorageTests.m]: Empty line before comment

File: AppCenter/AppCenter/Internals/Storage/MSDBStoragePrivate.h [hunk 4]
```diff
@@ -46,6 +46,31 @@ static const long kMSDefaultDatabaseSizeInBytes = 10 * 1024 * 1024;
...
+ * Query the number of pages (i.e.: SQLite "page_count") of the database.
+ *
+ * @param db Database handle.
+ *
+ * @return The number of pages.
+ */
++ (long)getPageCountInOpenedDatabase:(void *)db;
+
+/**
+ * Query the number of pages (i.e.: SQLite "page_size") of the database.
```
Comment (annakocheshkova) [inline AppCenter/AppCenter/Internals/Storage/MSDBStoragePrivate.h]: Please fix the comment

File: AppCenter/AppCenter/Internals/Storage/MSDBStorage.m [hunk 5]
```diff
@@ -289,17 +292,17 @@ + (sqlite3 *)openDatabaseAtFileURL:(NSURL *)fileURL withResult:(int *)result {
 
 + (long)getPageSizeInOpenedDatabase:(void *)db {
   NSArray<NSArray *> *rows = [MSDBStorage executeSelectionQuery:@"PRAGMA page_size;" inOpenedDatabase:db];
-  return [(NSNumber *)rows[0][0] longValue];
+  return rows.count > 0 && rows[0].count > 0 ? [(NSNumber *)rows[0][0] longValue] : 0;
```
Comment (ElektrojungeMS) [inline AppCenter/AppCenter/Internals/Storage/MSDBStorage.m]: I think that this is something that we can move to a helper method so we don't have the same line of code in multiple places.

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,6 +5,7 @@
 ### AppCenter
 
 * **[Fix]** Fix a possible deadlock if the SDK is started from a background thread.
+* **[FIX]** Fix possible crash if database query failed.
```
Comment (jaelim-ms) [inline CHANGELOG.md]: Fix **a** possible

Comment (MatkovIvan) [reply CHANGELOG.md]: It will be part of the next release

Comment (MatkovIvan) [reply CHANGELOG.md]:
```suggestion
* **[Fix]** Fix a crash if database query failed. 
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,11 @@
 # App Center SDK for iOS and macOS Change Log
 
+## Version 1.13.2
```
Comment (MatkovIvan) [inline CHANGELOG.md]:
```suggestion
## Version 1.13.2 (Not yet released)
```

File: AppCenter/AppCenter/Internals/Storage/MSDBStorage.m [hunk 8]
```diff
@@ -111,12 +111,12 @@ + (BOOL)tableExists:(NSString *)tableName inOpenedDatabase:(void *)db {
   NSString *query =
       [NSString stringWithFormat:@"SELECT COUNT(*) FROM \"sqlite_master\" WHERE \"type\"='table' AND \"name\"='%@';", tableName];
   NSArray<NSArray *> *result = [MSDBStorage executeSelectionQuery:query inOpenedDatabase:db];
-  return (result.count > 0) ? [(NSNumber *)result[0][0] boolValue] : NO;
+  return result.count > 0 && result[0].count > 0 ? [(NSNumber *)result[0][0] boolValue] : NO;
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Storage/MSDBStorage.m]: Can we use `querySingleValue` here?

Comment (jaelim-ms) [reply AppCenter/AppCenter/Internals/Storage/MSDBStorage.m]: It's returning different type. I guess that's why @dtrabo didn't change this one.

**Pull request merged**